### DISCUSSION TITLE
upload or select an already existing world

### DISCUSSION THREAD

**Discussion by dyyce** [Didstopia/valheim-server#3] [Q&A]
Hey,

Thanks for this docker image. I followed the guide and it worked out.
One question I have though is how to upload or select an already existing world?
I have uploaded my world into the `valheim-server/valheim_data/saves/worlds` dir. But don't know which param to set this as the selected world when the server starts.

cheers,

Taras

**Comment by dyyce** [comment]
So I have seen the env `VALHEIM_SERVER_WORLD`. I have a world in the dir called `myworld`. when I set `VALHEIM_SERVER_WORLD="myworld"` it does not take the existing world, but creates a new one callned `myworld_`.
Any idea how to solve this?

**Reply by Dids** [reply]
Good catch! This is a bug that comes from normalizing the world name. I'll get it fixed asap and will update here when it's live, just need to gracefully handle worlds that already have the `_` suffix, possibly by just renaming any worlds with the suffix to one without it.

**Reply by dyyce** [reply]
awesome! looking forward :)

**Reply by Dids** [reply]
Can you pull the latest image and give it a go? Should now migrate the files, but only one they don't already exist without the _ suffix.

**Reply by dyyce** [reply]
Hey @Dids,

yes, it is taking the world I have uploaded :)
Unfortunately, all the buildings are gone. Yet the map and fog of war state are from my local world.
But I don't think this has anything to do with the script you've provided. Might be some settings that I need to set.

Thank you very much for your support :)

**Reply by Dids** [reply]
That is unfortunate to hear. :(

Let me know if anything pops up, or if it turns out to be an issue with this container! My best guess is that since the world name technically changed, so did the seed etc. which did some kind of weird partial reset, but that's just a guess right now.

EDIT: Seeing this myself too, except my entire map seems to be reset. Could probably use my backup files and append the `_` suffix to get the previous world and such, but luckily this bug shouldn't be affecting any new users anymore. Just no idea how many it will affect as of now.

EDIT 2: Yup, copied my backups to `<world name>_.<ext>`, then set the world name env var to also include the underscore (`<worldname>_`) and it loads everything back up normally.

**Reply by dyyce** [reply]
Hey @Dids ,

Thank you, this worked!
During the setup, I renamed the world, added underscore etc. and that was the reason why the buildings were gone, while the map was the same. I re-uploaded my world again without renaming it and then it worked :)

Cheers,
Taras
